### DISCUSSION TITLE
Questions about vulnerabilities: CVE-2020-5208

### DISCUSSION THREAD

**Discussion by robinnie** [ipmitool/ipmitool#343] [Q&A]
Hello,

The problem is fixed on the master branch of ipmitool project since commit https://github.com/ipmitool/ipmitool/commit/7ccea283dd62a05a320c1921e3d8d71a87772637.
And It seems 6 commits as below for the fix of CVE-2020-5208. 
---------------------------------------------------------------------------
          commit e824c23316ae50beb7f7488f2055ac65e8b341f2
          Author: Chrostoper Ertl [chertl@microsoft.com](mailto:chertl@microsoft.com)
          Date:   Thu Nov 28 16:33:59 2019 +0000
          
          commit 840fb1cbb4fb365cb9797300e3374d4faefcdb10
          Author: Chrostoper Ertl [chertl@microsoft.com](mailto:chertl@microsoft.com)
          Date:   Thu Nov 28 16:44:18 2019 +0000
          
          commit 41d7026946fafbd4d1ec0bcaca3ea30a6e8eed22
          Author: Chrostoper Ertl [chertl@microsoft.com](mailto:chertl@microsoft.com)
          Date:   Thu Nov 28 16:51:49 2019 +0000
          
          commit 9452be87181a6e83cfcc768b3ed8321763db50e4
          Author: Chrostoper Ertl [chertl@microsoft.com](mailto:chertl@microsoft.com)
          Date:   Thu Nov 28 16:56:38 2019 +0000
          
          commit d45572d71e70840e0d4c50bf48218492b79c1a10
          Author: Chrostoper Ertl [chertl@microsoft.com](mailto:chertl@microsoft.com)
          Date:   Thu Nov 28 17:06:39 2019 +0000
          
          commit 7ccea283dd62a05a320c1921e3d8d71a87772637
          Author: Chrostoper Ertl [chertl@microsoft.com](mailto:chertl@microsoft.com)
          Date:   Thu Nov 28 17:13:45 2019 +0000
-----------------------------------------------------------------------
We want to backport the fix to our current version 1.8.15.
We have two concerns:
1.	Is there any missed commits except the above 6 commits?
2.	How to verify it to check if the issue is removed?

Thanks a lot in advance!
BR/Robin Nie

**Comment by AlexanderAmelkin** [comment]
The issues were quite hypothetic and no exploit has been presented, so I don't know how you could check except for modelling the situation described in the CVE. There should be no commits related to the CVE and not marked with the CVE-2020-5208.

**Reply by robinnie** [reply]
Thank you very much for your reply.

We reviewed the code change and found there is a memory leak issue on code change of commit: 840fb1

---
if(rsp->data_len < 1
		   || len > rsp->data_len - 1
		   || len > fru.size - offset)
		{
			printf(" Not enough buffer size");
                        **// here, need free buffer**
			return -1;
		}
----

One more question,  when will we have the release of 1.8.19? 

Thanks,
BR/Robin

**Reply by AlexanderAmelkin** [reply]
The bug you mentioned has been fixed in commit b97a1107 over two years ago and is a part of release [1.8.19](https://github.com/ipmitool/ipmitool/releases/tag/IPMITOOL_1_8_19)

**Comment by AlexanderAmelkin** [comment]
Just wanted to notice that 1.8.15 is outdated beyond any imagination. You should definitely switch to 1.8.19 and not backport parts of it to 1.8.15.

**Reply by robinnie** [reply]
Thank you very much!
